src/storage.c

```diff
@@ -0,0 +1,15 @@
+#include "storage.h"
+
+uint32_t get_secret() {
+    if (N_storage.secret == 0) {
+        // first access, generate and write to storage
+        return generate_secret();
+    }
+    return (volatile uint32_t) N_storage.secret;
+}
+
+uint32_t generate_secret() {
+    uint32_t new_secret = cx_rng_u32();
+    nvm_write((void *) &N_storage.secret, &new_secret, sizeof(uint32_t));
```

Comment:
Is it safe in Ledger to write secrets to nvm like this? Won't other apps be able to read the secret from the storage?

Reply:
Ok, I've been informed that this is related to https://github.com/HathorNetwork/hathor-ledger-app/issues/28

Reply:
That's a very good point.

Reply:
This is the only way to write to the persistent storage. Even if this secret is "leaked" to other apps they cannot infer the signature (that this secret will generate) because it uses a derivation path specific to our app (the OS prevents derivation outside your app predefined path prefix).

Reply:
But do other apps have access to our app's memory region?

Reply:
The OS should prevent this, but I couldn't find anything about read protections on the persistent storage.

Reply:
Should we ask on their discord server?